src/compiler/checker.ts

```diff
@@ -13192,6 +13211,9 @@ namespace ts {
                         checkAsyncFunctionReturnType(<FunctionLikeDeclaration>node);
                     }
                 }
+                if (node.kind === SyntaxKind.ConstructSignature || node.kind === SyntaxKind.ConstructorType) {

```

Comment:
I have already tried this. But checkSignatureDeclaration is also called from CheckConstructorDeclartion and checkFunctionOrMethodDeclaration (before their body is processed). So, we should put them in the condition